src/main/java/dev/codesoapbox/dummy4j/dummies/internet/UrlBuilder.java

```diff
@@ -0,0 +1,352 @@
+package dev.codesoapbox.dummy4j.dummies.internet;
+
+import dev.codesoapbox.dummy4j.Dummy4j;
+import dev.codesoapbox.dummy4j.exceptions.UrlCouldNotBeCreatedException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.StringJoiner;
+
+/**
+ * Generates randomized urls.
+ * <p>
+ * Default values:
+ * <p>
+ * <ul>
+ *  <li>port: -1 (no port)</li>
+ *  <li>protocol: https</li>
+ * </ul>
+ */
+public final class UrlBuilder {
+
+    /**
+     * Lower boundary for randomly generated url port
+     */
+    public static final int MIN_PORT = 1000;
+
+    /**
+     * Upper boundary for randomly generated url port
+     */
+    public static final int MAX_PORT = 9999;
+
+    /**
+     * Default port value (must be compatible with the URL class)
+     */
+    public static final int DEFAULT_PORT = -1;
+
+    /**
+     * Points to a list of possible top level domains
+     */
+    public static final String DEFAULT_TOP_LEVEL_DOMAIN_KEY = "#{internet.top_level_domain}";
+
+    /**
+     * Points to a list of popular top level domains
+     */
+    public static final String POPULAR_TOP_LEVEL_DOMAIN_KEY = "#{internet.popular_top_level_domain}";
+
+    /**
+     * Points to a list of country top level domains
+     */
+    public static final String COUNTRY_TOP_LEVEL_DOMAIN_KEY = "#{internet.country_top_level_domain}";
+
+    /**
+     * Points to a list of generic top level domains
+     */
+    public static final String GENERIC_TOP_LEVEL_DOMAIN_KEY = "#{internet.generic_top_level_domain}";
+
+    /**
+     * Points to a list of random words that are used as domains
+     */
+    public static final String ROOT_DOMAIN_KEY = "#{internet.param}";
+
+    /**
+     * Points to a list of random words that are used as parameter names
+     */
+    public static final String PARAM_KEY = "#{internet.root_domain}";
+
+    /**
+     * Points to a list of random words that are used as parameter values
+     */
+    public static final String PARAM_VALUE_KEY = "#{internet.root_domain}";
+
+    private static final String FILE_EXTENSION = ".html";
+    private static final String PATH_DELIMITER = "/";
+    private static final int FILE_NAME_LENGTH = 10;
+
+    private final Dummy4j dummy4j;
+
+    private List<UrlProtocol> possibleProtocols = Collections.singletonList(UrlProtocol.HTTPS);
+    private boolean withQueryParams;
```

Comment:
Just noticed this boolean. It's not necessary - you can just check if `howManyParams > 0`.